### ISSUE TITLE
Google cloud storage modification time update doesn't work anymore on v1.49.0

### ISSUE THREAD

**Issue by wmuizelaar** [rclone/rclone#3493]
#### What is the problem you are having with rclone?
I updated to the latest version, using the google-cloud-storage backend. It now doesn't have permissions anymore to update modification times on the remote:

`googleapi: Error 403: Insufficient Permission, insufficientPermissions`

#### What is your rclone version (output from `rclone version`)
```
rclone v1.49.0
- os/arch: darwin/amd64
- go version: go1.12.9
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Mac OS 10.14.6, 64 bits. 

####  Which cloud storage system are you using? (eg Google Drive)
Google Cloud Storage

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
`/usr/local/bin/rclone sync /tmp/backupmount  google-cloud-storage:backup-prive-laptop --fast-list --create-empty-src-dirs --config /Users/wmuizelaar/.config/rclone/rclone.conf --exclude-from /etc/rsyncd.exclude -l --delete-excluded --delete-during --transfers 100 --checkers 50 --buffer-size=256M --retries=1 -v --stats-one-line`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
An excerpt is:
```
2019/08/28 16:16:31 ERROR : usr/local/Homebrew/.git/FETCH_HEAD: Failed to set modification time: googleapi: Error 403: Insufficient Permission, insufficientPermissions
```
No further details are added unfortunately.

#### Remark
Could it be related to tightening the oauth-scopes as done in https://github.com/rclone/rclone/commit/9f4589a997c355f7946db1535da75424b7536d6f#diff-5942d78c896367baea86a01d94585146 ?

**Comment by ncw** [comment]
> Could it be related to tightening the oauth-scopes as done in 9f4589a#diff-5942d78c896367baea86a01d94585146 ?

Yes that is the cause.  According to my tests the `full_control` scope is needed to issue the metadata PATCH command.

So going down from "https://www.googleapis.com/auth/devstorage.read_write" to "https://www.googleapis.com/auth/devstorage.full_control" has meant that we can't run PATCH on their objects to set new metadata.

I removed this in the oauth review because Google said

> Thank you for submitting an OAuth App Verification request for the following Cloud scopes:
>
> - https://www.googleapis.com/auth/devstorage.full_control 
> - https://www.googleapis.com/auth/cloud-platform  
>
> Three-legged  OAuth2 scope grants are intended for human users to grant access to all  of their data hosted on a particular API.  Access to your requested  OAuth2 scopes would provide overly broad access for Google Cloud  Platform customers. Google Cloud Platform only supports grants to  specific resources for specific users/services, with access controlled  using Cloud IAM Policies.

Unfortunately I failed to check updating the mod time.

There is a work-around for this - do the oauth again with v1.48 (use `rclone config` and edit the remote). The config file and generated token will work fine with v1.49.  However the oauth with v1.48 will give nasty errors.

Here is the failing transaction

```
2019/08/28 10:15:58 DEBUG : HTTP REQUEST (req 0xc00013a400)
2019/08/28 10:15:58 DEBUG : PATCH /storage/v1/b/rclone-test-new/o/hello.txt?alt=json&prettyPrint=false HTTP/1.1
Host: www.googleapis.com
User-Agent: rclone/v1.49.0-DEV
Content-Length: 107
Authorization: XXXX
Content-Type: application/json
X-Goog-Api-Client: gl-go/1.11.0 gdcl/20190802
Accept-Encoding: gzip

{"bucket":"rclone-test-new","metadata":{"mtime":"2019-08-28T10:15:58.474594679+01:00"},"name":"hello.txt"}

2019/08/28 10:15:58 DEBUG : HTTP RESPONSE (req 0xc00013a400)
2019/08/28 10:15:58 DEBUG : HTTP/1.1 403 Forbidden
Content-Length: 160
Alt-Svc: quic=":443"; ma=2592000; v="46,43,39"
Content-Type: application/json; charset=UTF-8
Date: Wed, 28 Aug 2019 09:15:58 GMT
Server: UploadServer
Vary: Origin
Vary: X-Origin
Www-Authenticate: Bearer realm="https://accounts.google.com/"
X-Guploader-Uploadid: AEnB2UqIXx6XKDSerbe27ww7sd75QPdhsXfO_vGkFMnbY6d3nm_JdbrgzcmUeMhQAU96nE2eaBX1V4rx25elmri0woaliUMThQ

{"error":{"errors":[{"domain":"global","reason":"insufficientPermissions","message":"Insufficient Permission"}],"code":403,"message":"Insufficient Permission"}}


```

**Comment by ncw** [comment]
I've made a temporary fix for this here

https://beta.rclone.org/branch/v1.49.0-007-g676eb6f1-fix-3493-gcs-scope-beta/ (uploaded in 15-30 mins)

Can you give this a go?  If you are using a service account then it should just work.  If you are using oauth, then you will need to reauth `rclone config reconnect remote:`

**Comment by ncw** [comment]
OK here is a better fix which doesn't require elevated permissions at all

    Before this change we used PATCH on the object to update the metadata.
    
    Apparently this requires the "full_control" scope which Google were
    unhappy with in their oauth review.
    
    This changes it to update the metadata by copying the object ontop of
    itself (which is the way s3 works).  This can be done with normal
    permissions.

Can you give this a try please?

https://beta.rclone.org/branch/v1.49.0-007-g10c3b591-fix-3493-gcs-scope-copy-beta/ (uploaded in 15-30 mins)

**Comment by wmuizelaar** [comment]
Yes, this works like a charm!
```2019/08/29 18:26:47 INFO  : usr/local/Homebrew/.git/FETCH_HEAD: Updated modification time in destination```

**Comment by wmuizelaar** [comment]
Thanks for the quick respons and fix!

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.50

I might put this in a point release also
